spec/machine/AARCH64/MachineOps.thy

```diff
@@ -485,5 +485,32 @@ consts' setVSpaceRoot_impl :: "paddr \<Rightarrow> machine_word \<Rightarrow> un
 definition setVSpaceRoot :: "paddr \<Rightarrow> machine_word \<Rightarrow> unit machine_monad" where
   "setVSpaceRoot pt asid \<equiv> machine_op_lift $ setVSpaceRoot_impl pt asid"
 
+
+subsection "Secure Monitor Calls (SMC)"
+
+definition numSMCRegs :: nat where
+  "numSMCRegs \<equiv> 8"
+
+(* Tempting to use a list, but encoding the length requirement would make it ugly, and numSCMRegs
+   is only going to change when the SMC spec changes. *)
+type_synonym 'a eight_tuple = "'a \<times> 'a \<times>'a \<times>'a \<times>'a \<times>'a \<times>'a \<times>'a"
```

Comment:
might want to add spaces after the `×` consistently, I was staring at this for a moment thinking what `x'a` is